tests/teststun.nim

```diff
@@ -0,0 +1,87 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+{.used.}
+
+import options, strutils
+import bearssl
+import ../webrtc/udp_connection
+import ../webrtc/stun/stun_connection
+import ../webrtc/stun/stun_message
+import ../webrtc/stun/stun_attributes
+import ../webrtc/stun/stun_utils
+import ./asyncunit
+
+proc newRng(): ref HmacDrbgContext =
+  HmacDrbgContext.new()
+
+suite "Stun message encoding/decoding":
+  test "Get BindingRequest + encode & decode with a set username":
+    let
+      udpConn = UdpConn.init(AnyAddress)
+      conn = StunConn.init(udpConn, TransportAddress(AnyAddress), true, newRng())
+      msg = conn.getBindingRequest(username = "Test")
+      encoded = msg.encode(msg.getAttribute(AttrUsername))
+      decoded = StunMessage.decode(encoded)
+
+    check:
+      msg.msgType == decoded.msgType
+      msg.transactionId == decoded.transactionId
+      msg.getAttribute(AttrUsername) == decoded.getAttribute(AttrUsername)
+      msg.getAttribute(AttrICEControlling).isSome()
+      msg.getAttribute(AttrICEControlled).isNone()
+      Priority.decode(msg.getAttribute(AttrPriority).get()).priority == 0x7effffff
+      # encoding adds Fingerprint as attributes
+      msg.attributes.len() == decoded.attributes.len() - 1
```

Comment:
should we check the fingerprint is calculated correctly?

Reply:
In this specific test, it does not seem really important. I added a check to see if it is present, though.
On another hand, according to the RFC:
```
   When the FINGERPRINT extension is used, an agent includes the
   FINGERPRINT attribute in messages it sends to another agent.
   Section 15.5 describes the placement and value of this attribute.
   When the agent receives what it believes is a STUN message, then, in
   addition to other basic checks, the agent also checks that the
   message contains a FINGERPRINT attribute and that the attribute
   contains the correct value.
 ```
 And I do not check this specifically. I'll add it.

Reply:
I added the Fingerprint check.